### PR TITLE
[Fix](merge-on-write) Fix UBSan error cause by member call on null pointer

### PR DESCRIPTION
**Pull request by bobhan1**
## Proposed changes

fix member call on null pointer in https://github.com/apache/doris/pull/38331
image

### CODE REVIEW

**Comment by bobhan1** [comment]
run buildall

File: be/src/olap/txn_manager.cpp [hunk 1]
```diff
@@ -747,8 +747,10 @@ void TxnManager::force_rollback_tablet_related_txns(OlapMeta* meta, TTabletId ta
             }
         }
     }
-    static_cast<void>(
-            RowsetMetaManager::remove_tablet_related_partial_update_info(meta, tablet_id));
+    if (meta != nullptr) {
+        static_cast<void>(
+                RowsetMetaManager::remove_tablet_related_partial_update_info(meta, tablet_id));
+    }
```
Comment (yiguolei) [inline be/src/olap/txn_manager.cpp:753]: why cast the result as static void?
If you do not need check the result, I think you should modify the method to return void.

**Comment by bobhan1** [comment]
run buildall

**Review by zhannngchen — APPROVED** [review]

**Comment by bobhan1** [comment]
run p1

**Comment by bobhan1** [comment]
run cloud_p1

**Comment by bobhan1** [comment]
run performance

**Review by dataroaring — APPROVED** [review]
LGTM

**Comment by bobhan1** [comment]
run performance

**Pull request merged**